src/main/java/io/stargate/sgv2/jsonapi/util/JsonUtil.java

```diff
@@ -0,0 +1,61 @@
+package io.stargate.sgv2.jsonapi.util;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+public class JsonUtil {
+  /**
+   * Method that compares to JSON values for equality using Mongo semantics which are otherwise same
+   * as for JSON but additionally require exact ordering of Object (sub-document) values.
+   *
+   * @param node1 First JSON value to compare
+   * @param node2 Second JSON value to compare
+   * @return True if JSON values are equal according to Mongo semantics; false otherwise.
+   */
+  public static boolean equalsOrdered(JsonNode node1, JsonNode node2) {
+    JsonNodeType type = node1.getNodeType();
+    if (node2.getNodeType() != type) {
+      return false;
+    }
+    switch (type) {
+      case ARRAY -> {
+        if (node1.size() != node2.size()) {
+          return false;
+        }
+        Iterator<JsonNode> it = node1.elements();
+        for (JsonNode value : node2) {
+          if (!equalsOrdered(value, it.next())) {
+            return false;
+          }
+        }
+        return true;
+      }
+      case OBJECT -> {
```

Comment:
very strange, you would add two same objects that have fields in different order.. I guess mongo does that, but its very strange

Reply:
@ivansenic I agree, it is weird. But it is Mongo semantics :-(